src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,156 @@
+
+        }
+
+        /// <summary>
+        /// Uses the specified protocol. The broker must be set before use.
+        /// </summary>
+        /// <param name="protocol">The protocol version to use</param>
+        public RabbitMqBinding(IProtocol protocol)
+        {
+            BrokerProtocol = protocol;
+            Name = "RabbitMQBinding";
+            Namespace = "http://schemas.rabbitmq.com/2007/RabbitMQ/";
+
+            Initialize();
+        }
+
+        public override BindingElementCollection CreateBindingElements()
+        {
+            //Transport.HostName = HostName;
+            //Transport.Port = Port;
+            Transport.BrokerProtocol = BrokerProtocol;
+            if (MaxMessageSize != DefaultMaxMessageSize)
+            {
+                Transport.MaxReceivedMessageSize = MaxMessageSize;
+            }
+            BindingElementCollection elements = new BindingElementCollection { _encoding, Transport, };
+
+            return elements;
+        }
+
+        private void Initialize()
+        {
+            lock (this)
+            {
+                if (!_isInitialized)
+                {
+                    Transport = new RabbitMqTransportBindingElement();
+                    _encoding = new TextMessageEncodingBindingElement();
+                    MaxMessageSize = DefaultMaxMessageSize;
+                    _isInitialized = true;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the scheme used by the binding, soap.amqp
+        /// </summary>
+        public override string Scheme
+        {
+            get { return CurrentVersion.Scheme; }
+        }
+
+        /// <summary>
+        /// Specifies the hostname of the RabbitMQ Server
+        /// </summary>
+        public string HostName { get; set; }
+
+        /// <summary>
+        /// Specifies the RabbitMQ Server port
+        /// </summary>
+        public int Port { get; set; }
+
+        /// <summary>
+        /// Specifies the maximum encoded message size
+        /// </summary>
+        public long MaxMessageSize { get; set; }
+
+        /// <summary>
+        /// Specifies the version of the AMQP protocol that should be used to communicate with the broker
+        /// </summary>
+        public IProtocol BrokerProtocol { get; set; }
+
+        /// <summary>
+        /// Gets the AMQP transport binding element
+        /// </summary>
+        public RabbitMqTransportBindingElement Transport { get; private set; }
+
+        /// <summary>
+        /// Determines whether or not the TransactionFlowBindingElement will
+        /// be added to the channel stack
+        /// </summary>
+        public bool TransactionFlow { get; set; }
+
+        /// <summary>
+        /// Specifies whether or not the CompositeDuplex and ReliableSession
+        /// binding elements are added to the channel stack.
+        /// </summary>
+        public bool OneWayOnly { get; set; }
+    }
```

Comment:
We don't have either CompositeDuplex or ReliableSession yet so remove or comment.